File: java-example/src/test/java/org/onflow/examples/java/verifySignature/userSignatureValidateAny/UserSignatureValidateAnyExampleTest.java
package org.onflow.examples.java.verifySignature.userSignatureValidateAny;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.FlowEmulatorProjectTest;
import org.onflow.flow.common.test.FlowTestAccount;
import org.onflow.flow.common.test.FlowTestClient;
import org.onflow.flow.common.test.TestAccount;
import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.cadence.BooleanField;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.fail;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
public class UserSignatureValidateAnyExampleTest {
    @FlowTestAccount
    private TestAccount testAccount;
    @FlowTestClient
    private FlowAccessApi accessAPI;
    private UserSignatureValidateAnyExample connector;

    @BeforeEach
    public void setup() {
        connector = new UserSignatureValidateAnyExample(accessAPI);
    }

    @Test
    public void canVerifyUserSignature() {
        try {
            BooleanField txResult = connector.verifyUserSignatureValidateAny(testAccount.getFlowAddress(), testAccount.getPrivateKey(), "ananas");

            if (txResult != null) {
                assertEquals(Boolean.TRUE, txResult.getValue(), "Signature verification failed");
            } else {
                fail("Signature verification failed, no transaction result received");
            }
        } catch (Exception e) {
            fail("Test failed with exception: " + e.getMessage());
        }
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/verifySiganture/userSignatureValidateAny/UserSignatureValidateAnyExampleTest.kt
package org.onflow.examples.kotlin.verifySiganture.userSignatureValidateAny

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Assertions.fail
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.examples.kotlin.verifySignature.userSignatureValidateAny.UserSignatureValidateAnyExample
import org.onflow.flow.common.test.*
import org.onflow.flow.sdk.*
import org.onflow.flow.sdk.cadence.BooleanField

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
internal class UserSignatureValidateAnyExampleTest {
    @FlowTestAccount
    lateinit var testAccount: TestAccount

    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var connector: UserSignatureValidateAnyExample

    @BeforeEach
    fun setup() {
        connector = UserSignatureValidateAnyExample(accessAPI)
    }

    @Test
    fun `Can verify user signature`() {
        val txResult = connector.verifyUserSignatureValidateAny(testAccount.flowAddress, testAccount.privateKey, "ananas")

        if (txResult is BooleanField) {
            assertTrue(txResult.value!!, "Signature verification failed")
        } else {
            fail("Expected BooleanField but got ${txResult::class.simpleName}")
        }
    }
}
